okapi-core/src/test/java/org/folio/okapi/ModuleTest.java

```diff
@@ -1241,12 +1250,21 @@ public void testSystemInterfaces(TestContext context) {
       .log().ifValidationFails()
       .extract().headers();
     final String locHdrEnable = headers.getValue("Location");
-    List<Header> list = headers.getList("X-Tenant-Perms-Result");
-    Assert.assertEquals(2, list.size()); // one for okapi, one for header-1
-    Assert.assertThat("okapi perm result",
-      list.get(0).getValue(), containsString("okapi.all"));
-    Assert.assertThat("header-1perm result",
-      list.get(1).getValue(), containsString("header-1"));
+    // one trace from Okapi, two from the header module since it's called twice
+    context.assertEquals(3, headers.getValues("X-Okapi-Trace").size());
+
+    String body = given()
+        .header("X-Okapi-Tenant", okapiTenant)
+        .get("/permResult")
+        .then()
+        .statusCode(200)
+        .log().ifValidationFails()
+        .extract().body().asString();
+
+    JsonArray ar = new JsonArray(body);
+    context.assertEquals(2, ar.size());
+    context.assertEquals("okapi-0.0.0", ar.getJsonObject(0).getString("moduleId"));
+    context.assertEquals("header-1", ar.getJsonObject(1).getString("moduleId"));
```

Comment:
```suggestion
    .body("$", hasSize(2))
    .body("[0].moduleId", is("okapi-0.0.0"))
    .body("[1].moduleId", is("header-1"));
```
RestAssured has JSON path (GPath syntax) support. Simply use a recent RestAssured version, put `RestAssured.defaultParser = Parser.JSON;` into setUp, and import the hamcrest matcher:
```
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.Matchers.is;
```

Reply:
yet another way of doing things.. But shorter, so I've done that.